{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "using BioSequences\n",
    "using Pseudoseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 700bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n",
      "┌ Info: - ✔ Subsampled pool at 70X coverage (590465 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of 250bp paired-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:66\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.001\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 825306 paired end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:214\n",
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n",
      "┌ Info: - ✔ Subsampled pool at 30X coverage (3163 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of single-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:69\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.1\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 3163 single end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:312\n",
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:49\n",
      "┌ Info: Attaching 1000000 tags randomly to each molecule in pool...\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/pool.jl:166\n",
      "┌ Info: Done\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/pool.jl:168\n",
      "┌ Info: - ✔ Created pool of tagged fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:51\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 700bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n",
      "┌ Info: - ✔ Subsampled pool at 50X coverage (421761 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of 250bp paired-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:66\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.1\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 594438 paired end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:214\n",
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 700bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n",
      "┌ Info: - ✔ Subsampled pool at 70X coverage (590465 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of 250bp paired-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:66\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.001\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 826668 paired end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:214\n",
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n",
      "┌ Info: - ✔ Subsampled pool at 30X coverage (3163 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of single-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:69\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.1\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 3163 single end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:312\n",
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:49\n",
      "┌ Info: Attaching 1000000 tags randomly to each molecule in pool...\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/pool.jl:166\n",
      "┌ Info: Done\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/pool.jl:168\n",
      "┌ Info: - ✔ Created pool of tagged fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:51\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 700bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n",
      "┌ Info: - ✔ Subsampled pool at 50X coverage (421761 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of 250bp paired-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:66\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.1\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 595424 paired end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:214\n",
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 700bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n",
      "┌ Info: - ✔ Subsampled pool at 70X coverage (590465 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of 250bp paired-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:66\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.001\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 826462 paired end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:214\n",
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n",
      "┌ Info: - ✔ Subsampled pool at 30X coverage (3163 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of single-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:69\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.1\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 3163 single end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:312\n",
      "┌ Info: - ✔ Created pool of 5000 copies of a 4217614bp genome\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:45\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:49\n",
      "┌ Info: Attaching 1000000 tags randomly to each molecule in pool...\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/pool.jl:166\n",
      "┌ Info: Done\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/pool.jl:168\n",
      "┌ Info: - ✔ Created pool of tagged fragments with an average length of 40000bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:51\n",
      "┌ Info: - ✔ Created pool of fragments with an average length of 700bp\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "┌ Info: - ✔ Subsampled pool at 50X coverage (421761 molecules)\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:62\n",
      "┌ Info: - ✔ Created set of 250bp paired-end reads\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:66\n",
      "┌ Info: - ✔ Applied sequencing errors at a per-base rate of 0.1\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/sequence.jl:73\n",
      "┌ Info: - ✔ Wrote 595388 paired end reads to FASTQ file\n",
      "└ @ Pseudoseq /Users/ggarcia/.julia/packages/Pseudoseq/72NPA/src/reads.jl:214\n"
     ]
    },
    {
     "data": {
      "text/plain": []
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "595388 paired-end reads:\n",
      " Read length: 250\n",
      " Number of errors: 14884700\n"
     ]
    }
   ],
   "source": [
    "## Create some PE for all\n",
    "Pseudoseq.sequence(\"./p1/p1-yeast-chr45.fasta\", \"./p1/p1-pe-reads.fastq\", ng=5000, flen=700, cov=70, paired = true, rdlen = 250, err = 0.001)\n",
    "Pseudoseq.sequence(\"./p1/p1-yeast-chr45.fasta\", \"./p1/p1-long-reads.fastq\", ng = 5000, flen = 40000, cov = 30, rdlen = nothing, err = 0.1, paired = false)\n",
    "Pseudoseq.sequence(\"./p1/p1-yeast-chr45.fasta\", \"./p1/p1-link-reads.fastq\", ng = 5000, tusize = 1000000, taggedflen = 40000, flen = 700, cov = 50, rdlen = 250, err = 0.1)\n",
    "\n",
    "Pseudoseq.sequence(\"./p2/p2-yeast-chr45.fasta\", \"./p2/p2-pe-reads.fastq\", ng=5000, flen=700, cov=70, paired = true, rdlen = 250, err = 0.001)\n",
    "Pseudoseq.sequence(\"./p2/p2-yeast-chr45.fasta\", \"./p2/p2-long-reads.fastq\", ng = 5000, flen = 40000, cov = 30, rdlen = nothing, err = 0.1, paired = false)\n",
    "Pseudoseq.sequence(\"./p2/p2-yeast-chr45.fasta\", \"./p2/p2-link-reads.fastq\", ng = 5000, tusize = 1000000, taggedflen = 40000, flen = 700, cov = 50, rdlen = 250, err = 0.1)\n",
    "\n",
    "Pseudoseq.sequence(\"./child/child-yeast-chr45.fasta\", \"./child/child-pe-reads.fastq\", ng=5000, flen=700, cov=70, paired = true, rdlen = 250, err = 0.001)\n",
    "Pseudoseq.sequence(\"./child/child-yeast-chr45.fasta\", \"./child/child-long-reads.fastq\", ng = 5000, flen = 40000, cov = 30, rdlen = nothing, err = 0.1, paired = false)\n",
    "Pseudoseq.sequence(\"./child/child-yeast-chr45.fasta\", \"./child/child-link-reads.fastq\", ng = 5000, tusize = 1000000, taggedflen = 40000, flen = 700, cov = 50, rdlen = 250, err = 0.1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Julia 1.1.0",
   "language": "julia",
   "name": "julia-1.1"
  },
  "language_info": {
   "file_extension": ".jl",
   "mimetype": "application/julia",
   "name": "julia",
   "version": "1.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
